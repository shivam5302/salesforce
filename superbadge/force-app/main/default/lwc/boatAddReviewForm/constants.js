
const SUCCESS = {
    Title: 'Success',
    Message: 'Review saved successfully!',
    Variant: 'success'
}

const ERROR = {
    Title: 'Error',
    Message: 'Error while saving the record!',
    Variant: 'error'
}

export {
    SUCCESS,
    ERROR
}