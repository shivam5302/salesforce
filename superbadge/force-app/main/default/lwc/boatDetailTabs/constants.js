
import ID_FIELD from '@salesforce/schema/Boat__c.Id'
import NAME_FIELD from '@salesforce/schema/Boat__c.Name';
import TYPE_FIELD from '@salesforce/schema/Boat__c.BoatType__c';
import LENGTH_FIELD from '@salesforce/schema/Boat__c.Length__c';
import PRICE_FIELD from '@salesforce/schema/Boat__c.Price__c';
import Description_FIELD from '@salesforce/schema/Boat__c.Description__c';


const BOAT_FIELDS = {
	ID_FIELD,
	NAME_FIELD,
	TYPE_FIELD,
	LENGTH_FIELD,
	PRICE_FIELD,
	Description_FIELD
};

export {
	BOAT_FIELDS
}