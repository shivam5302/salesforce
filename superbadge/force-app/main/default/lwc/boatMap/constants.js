
const SELECT_BOAT_LABEL = "Please select a boat to see its location!";

export {
	SELECT_BOAT_LABEL
}