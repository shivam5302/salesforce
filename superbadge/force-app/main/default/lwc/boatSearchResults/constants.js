


export const COLUMNS = [
    { label: 'Name', fieldName: 'Name', editable: true },
    { label: 'Length', fieldName: 'Length__c', editable: true },
    { label: 'Price', fieldName: 'Price__c', editable: true },
    { label: 'Description', fieldName: 'Description__c', editable: true }
];

export const TOAST = {
    SUCCESS: {
        VARIANT: 'success',
        TITLE: 'Success',
        MESSAGE: 'Data saved successfully',
    },
    ERROR: {
        VARIANT: 'error',
        TITLE: 'Error',
        MESSAGE: 'Error while saving data {0}'
    }

}