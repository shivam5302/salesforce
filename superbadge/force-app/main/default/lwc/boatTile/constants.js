const TILE_WRAPPER_SELECTED_CLASS = 'tile-wrapper selected';
const TILE_WRAPPER_UNSELECTED_CLASS = 'tile-wrapper'

export {
    TILE_WRAPPER_SELECTED_CLASS,
    TILE_WRAPPER_UNSELECTED_CLASS
}