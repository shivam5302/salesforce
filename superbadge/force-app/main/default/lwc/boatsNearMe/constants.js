const LABEL_YOU_ARE_HERE = 'You are here!';
const ICON_STANDARD_USER = 'standard:user';
const ERROR_TITLE = 'Error loading Boats Near Me';
const ERROR_VARIANT = 'error';

export {
	LABEL_YOU_ARE_HERE,
	ICON_STANDARD_USER,
	ERROR_TITLE,
	ERROR_VARIANT
}