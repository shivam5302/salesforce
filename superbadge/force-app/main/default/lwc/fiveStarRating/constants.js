const EDITABLE_CLASS = "c-rating"
const READ_ONLY_CLASS = "readonly c-rating"

export {
    EDITABLE_CLASS,
    READ_ONLY_CLASS
}