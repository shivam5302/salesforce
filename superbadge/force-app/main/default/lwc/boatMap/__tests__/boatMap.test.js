import { createElement } from 'lwc';
import BoatMap from 'c/boatMap';
import { getRecord } from 'lightning/uiRecordApi';
import { subscribe } from 'lightning/messageService';
import MESSAGE_CHANNEL from '@salesforce/messageChannel/BoatMessageChannel__c';
import { SELECT_BOAT_LABEL } from '../constants';

const flushPromises = async () => new Promise((resolve) => setTimeout(resolve, 0));

// Mock realistic data
const mockGetRecord = require('./data/getRecord.json');

jest.mock('lightning/messageService', () => ({
	subscribe: jest.fn(),
	unsubscribe: jest.fn(),
	MessageContext: jest.fn(),
}));

jest.mock('lightning/uiRecordApi',
	() => {
		const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
		return {
			getRecord: createApexTestWireAdapter(jest.fn()),
		};
	},
	{ virtual: true }
)

describe('c-boat-map', () => {
	let element;

	beforeEach(() => {
		element = createElement('c-boat-map', {
			is: BoatMap
		});
		// element.recordId = 'testId';
		document.body.appendChild(element);
		
		const mockMessage = { recordId: 'abc123' };
		const messageCallback = subscribe.mock.calls[0][2];
		messageCallback(mockMessage);
	});

	afterEach(() => {
		// Reset the DOM to clean up between tests
		while (document.body.firstChild) {
			document.body.removeChild(document.body.firstChild);
		}
		subscribe.mockClear();
	});

	it('Should show label no boat selected', () => {
		const span = element.shadowRoot.querySelector("span");
		expect(span.textContent).toBe(SELECT_BOAT_LABEL);
		expect(element).toMatchSnapshot();
	});

	it('Wire Should show locations on map', async () => {
		// Given
		getRecord.emit(mockGetRecord);

		// element.recordId='123'
		debugger;
		// Wait for promises to resolve
		await flushPromises();

		const map = element.shadowRoot.querySelector("lightning-map");

		// Then
		expect(map).toBeDefined();
		expect(element).toMatchSnapshot();
		expect(element.recordId).toBe('abc123');
	});

	it('Wire Should return error', async () => {
		// Given
		const mockError = { message: 'Record not found' };
		getRecord.error(mockError);

		// Wait for promises to resolve
		await flushPromises();

		// Then
		expect(element).toMatchSnapshot();
	});
});
