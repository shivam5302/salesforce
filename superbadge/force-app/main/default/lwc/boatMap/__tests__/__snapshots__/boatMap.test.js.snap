// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-boat-map Should show label no boat selected 1`] = `
<c-boat-map>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <span
        class="slds-align_absolute-center"
      >
        Please select a boat to see its location!
      </span>
    </lightning-card>
</c-boat-map>
`;

exports[`c-boat-map Wire Should return error 1`] = `
<c-boat-map>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <div
        class="slds-theme_error"
      >
        <lightning-icon>
          #shadow-root(open)
        </lightning-icon>
        <span
          class="slds-p-left_xx-small"
        >
          Some error message
        </span>
      </div>
    </lightning-card>
</c-boat-map>
`;

exports[`c-boat-map Wire Should show locations on map 1`] = `
<c-boat-map>
  #shadow-root(open)
    <lightning-card>
      #shadow-root(open)
        <slot />
        <slot
          name="actions"
        />
        <slot
          name="footer"
        />
        <slot
          name="title"
        />
      <lightning-map>
        #shadow-root(open)
      </lightning-map>
    </lightning-card>
</c-boat-map>
`;
